app/src/main/java/com/google/maps/android/ktx/demo/components/ScaleBar.kt

```diff
@@ -0,0 +1,289 @@
+                metricDistance /= 1000
+            }
+
+            var imperialUnits = "ft"
+            var imperialDistance = toFeet(horizontalLineWidth.toDouble())
+            if (imperialDistance > 5280) {
+                // Switch from ft to miles as unit
+                imperialUnits = "mi"
+                imperialDistance = toMiles(imperialDistance)
+            }
+
+            ScaleText(
+                modifier = Modifier.align(End),
+                textColor = textColor,
+                shadowColor = shadowColor,
+                text = "${imperialDistance.toInt()} $imperialUnits"
+            )
+            ScaleText(
+                modifier = Modifier.align(End),
+                textColor = textColor,
+                shadowColor = shadowColor,
+                text = "$metricDistance $metricUnits"
+            )
+        }
+    }
+}
+
+@Composable
+private fun ScaleText(
+    modifier: Modifier = Modifier,
+    text: String,
+    textColor: Color = DarkGray,
+    shadowColor: Color = Color.White,
+) {
+    Text(
+        text = text,
+        fontSize = 12.sp,
+        color = textColor,
+        textAlign = TextAlign.End,
+        modifier = modifier,
+        style = MaterialTheme.typography.h4.copy(
+            shadow = Shadow(
+                color = shadowColor,
+                offset = Offset(2f, 2f),
+                blurRadius = 1f
+            )
+        )
+    )
+}
+
+/**
+ * An animated scale bar that appears when the scale of the map changes, and then disappears
+ * after [visibilityTimeoutMs]. This composable wraps [ScaleBar] with visibility animations.
+ *
+ * To use this function, pass in a [GoogleMap], which the [DisappearingScaleBar] will use for the
+ * projection of current map view to meters by registering to [GoogleMap.cameraMoveEvents]. Note
+ * that using this will override an existing listener (if any) to
+ * [GoogleMap.setOnCameraMoveListener].
+ *
+ * To avoid overriding existing listeners of [GoogleMap.setOnCameraMoveListener], implement your own
+ * observer on camera move events (e.g., val projection: Projection by googleMap.cameraMoveEvents()
+ * .collectAsState(googleMap.projection)) and animations and pass in the [Projection] to the
+ * relevant [ScaleBar] constructor.
+ */
+@ExperimentalCoroutinesApi
+@Composable
+fun DisappearingScaleBar(
+    modifier: Modifier = Modifier,
+    googleMap: GoogleMap,
+    textColor: Color = DarkGray,
+    lineColor: Color = DarkGray,
+    shadowColor: Color = Color.White,
+    visibilityTimeoutMs: Long = 3000
+) {
+    val visible = remember {
+        MutableTransitionState(true)
+    }
+
+    val projection: Projection by googleMap.cameraMoveEvents()
+        .collectAsState(googleMap.projection)
+
+    LaunchedEffect(key1 = projection) {
+        if (visible.isIdle && !visible.currentState) {
+            Log.d("scale-bar", "show")
+            // FIXME - Why doesn't the ScaleBar re-appear on the map when this executes?
```

Comment:
I think this is because `visible` is not using property delegation so state changing isn't triggering recomposition as expected. I suggested an edit above—does that fix the issue?

Reply:
>I think this is because visible is not using property delegation so state changing isn't triggering recomposition as expected.

Ah, thanks, that's a good observation.

>I suggested an edit above—does that fix the issue?

Unfortunately no - this results in the build error:

```
Type 'TypeVariable(T)' has no method 'getValue(Nothing?, KProperty<*>)' and thus it cannot serve as a delegate
```

I also tried:

```kotlin
    val visible by remember {
        mutableStateOf(MutableTransitionState(true))
    }
```

...which builds, but no change when executed.

FWIW, the `MutableTransitionState` docs:
https://developer.android.com/reference/kotlin/androidx/compose/animation/core/MutableTransitionState

...use `...= remember { MutableTransitionState(false) }` but that may be because it's a one-off animation (?). Unfortunately there doesn't seem to be a whole lot of documentation about this. 

I'll take a look at the Compose samples to see if I can find anything there.

Reply:
Actually, something else is going on here. Looking at the diff that supposedly fixed the above issue:
https://android.googlesource.com/platform/frameworks/support/+/99632f58d6e8f7c10db61bdd9158b3eb5f8e48f9%5E%21/#F2

...it provided new defaults for `enter` and `exit`:

```
 fun AnimatedVisibility(
     visibleState: MutableTransitionState<Boolean>,
     modifier: Modifier = Modifier,
-    enter: EnterTransition,
-    exit: ExitTransition,
+    enter: EnterTransition = fadeIn() + expandIn(),
+    exit: ExitTransition = fadeOut() + shrinkOut(),
     content: @Composable() AnimatedVisibilityScope.() -> Unit
 ) {
```

...which matches the source I see if I look at AnimatedVisibility I'm using as a dependency in Android Studio.

Instead, it seems something is going on with the default `EnterTransition` of `fadeIn() + expandIn()` that just doesn't work here. If I specify this same animation myself as well explicitly it doesn't work. But if I use:

```kotlin
    AnimatedVisibility(
        visibleState = visible,
        enter = expandHorizontally() + fadeIn(),
        exit = shrinkHorizontally() + fadeOut(),
    ) {
```

...this `enter` animation works.

Reply:
From more testing, a simple `fadeIn()` and `fadeOut()` work, which also matches the behavior of the scale in Google Maps. I switched to this in https://github.com/googlemaps/android-maps-ktx/pull/175/commits/871bc5afb9f365ca6d14db1ae22e2a8c47d1cb61, which also introduces `enterTransition` and `exitTransition` to method signature to allow user to specify these animations.